packages/expo-web-browser/ios/WebBrowserSession.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import SafariServices
+import ExpoModulesCore
+
+internal class WebBrowserSession: NSObject, SFSafariViewControllerDelegate {
+  let viewController: SFSafariViewController
+  var modalNavigationController: UINavigationController?
+  var promise: Promise?
+  var isOpen: Bool {
+    promise != nil
+  }
+
+  init(url: URL, options: WebBrowserOptions) {
+    let configuration = SFSafariViewController.Configuration()
+    configuration.barCollapsingEnabled = options.enableBarCollapsing
+    configuration.entersReaderIfAvailable = options.readerMode
+
+    viewController = SFSafariViewController(url: url, configuration: configuration)
+    viewController.dismissButtonStyle = options.dismissButtonStyle.toSafariDismissButtonStyle()
+    viewController.preferredBarTintColor = options.toolbarColor
+    viewController.preferredControlTintColor = options.controlsColor
+
+    super.init()
+    viewController.delegate = self
+
+    // By setting the modal presentation style to OverFullScreen, we disable the "Swipe to dismiss"
+    // gesture that is causing a bug where sometimes `safariViewControllerDidFinish` is not called.
+    // There are bugs filed already about it on OpenRadar.
```

Comment:
Maybe add a references to the bugs here? 🤔

Reply:
If only I have the references 😂 I just kept this comment and logic with the hope I'll address it later. There were several things that I was not sure about and would rather like to test/fix them one by one.